Content.Server/Construction/ConstructionSystem.cs

```diff
@@ -97,6 +97,12 @@ private IEnumerable<IEntity> EnumerateNearby(IEntity user)
             // We need a place to hold our construction items!
             var container = ContainerHelpers.EnsureContainer<Container>(user, materialContainer, out var existed);
 
+            if (user.IsInContainer())
+            {
+                user.PopupMessageCursor(Loc.GetString("construction-system-inside-container"));
+                return null;
+            }
```

Comment:
You should do this before ensuring the container

Reply:
oh my yes